dll/win32/shell32/dialogs/recycler_prop.cpp

```diff
@@ -146,6 +147,11 @@ InitializeRecycleBinDlg(HWND hwndDlg, WCHAR DefaultDrive)
                                 defIndex = itemCount;
                                 pDefault = pItem;
                             }
+
+                            dwSize = sizeof(DWORD);
+                            RegGetValueW(HKEY_CURRENT_USER, L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Policies\\Explorer", L"ConfirmFileDelete", RRF_RT_DWORD | RRF_ZEROONFAILURE, NULL, &pItem->dwConfirmDelete, &dwSize);
```

Comment:
The policy doesn't check or uncheck the box. It "only" makes the user able to act on the checkbox.
The actual confirm-delete value is stored in a shell state value.